fix: 🐛 Session setup method deprecation message

calcaide opened the pull request:
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-6390)

## Description

Fix session setup method deprecation message. More info within the Jira ticket

Deprecation message in `addons/auth` before the changes:
![ember-simple-auth-before](https://user-images.githubusercontent.com/9775006/195349009-273e73e6-c6b0-4a56-8526-d2207b675058.jpg)

No deprecation message in `addons/auth` after the changes:
![ember-simple-auth-after](https://user-images.githubusercontent.com/9775006/195349092-cc3aabd1-3ac9-4d19-96d4-dd6bc0a31b59.jpg)

Deprecation message in `ui/admin` before changes:
<img width="1679" alt="adminui-simpleauth-before" src="https://user-images.githubusercontent.com/9775006/195554666-c8390cde-6eec-4704-b495-670dd615c93f.png">


NO deprecation message in `ui/admin` after changes:
<img width="1680" alt="desktop-simpleauth-after" src="https://user-images.githubusercontent.com/9775006/195554694-c129e3a6-a658-4e9d-bdfb-a12ac8daa202.png">

Deprecation message in `ui/desktop` before changes:
<img width="1680" alt="desktop-simpleauth-before" src="https://user-images.githubusercontent.com/9775006/195554808-d9063797-0ef8-468f-abd8-f880f6cddecb.png">


NO deprecation message in `ui/desktop` after changes:

<img width="1680" alt="desktop-simpleauth-after" src="https://user-images.githubusercontent.com/9775006/195554839-dee5ac8b-c5dc-4abf-b9e4-9dbc38a358f3.png">

ZedLi commented on addons/auth/ember-cli-build.js:
```diff
@@ -4,7 +4,10 @@ const EmberAddon = require('ember-cli/lib/broccoli/ember-addon');
   let app = new EmberAddon(defaults, {
-    // Add options here
+    'ember-simple-auth': {
+      // https://github.com/mainmatter/ember-simple-auth/blob/master/guides/upgrade-to-v4.md#call-sessionsetup-method-in-your-applicationroute
```
Don't we have to call `this.session.setup();` manually now in our application routes according to the docs to turn this feature on?

calcaide replied:
Answered this on a comment below 😉  thank you for the feedback!

calcaide commented:
@ZedLi @randallmorey 

Sorry I did not explain the plan for this PR.

My initial plan was to change JUST the auth addon. Then consumers (admin UI and DC) will start to show deprecation message, and later on I was going to provide a PR fix those consumers.

Following the plan this PR is working NP. The consumers start show those deprecation message. 

Be aware:
- The change on `addon/auth/ember-cli-build` was to tell `ember-simple-auth` we are going to setup session method ourselves. This forces `ember-simple-auth` to expect us manually call `session.setup()` aka prepare for the upgrade and because that deprecation message stops showing within `addon/auth`.
- Within `addon/auth` we can not call `this.session.setup()` because we do not even have basic routing defined, therefore no need to do so. We do not have an app that needs run either.

**That said your comments made me realize my strategy was too cumbersome. I updated this PR to achieve next:**
- Update `addon/auth `(done in the first iteration of this PR).
- Update consumers of auth Admin UI and Desktop Client UI to manually call `session.setup()` and change `ember-cli-build` to tell `ember-simple-auth` we will set it manually.

I did also updated screenshots, now showing consumers before/after changes.
